lib/parsers/yarn2-lock-parser.ts

```diff
@@ -41,8 +44,18 @@ export class Yarn2LockParser extends LockParserBase {
 
       Object.entries(rawYarnLock).forEach(
         ([fullDescriptor, versionData]: [string, any]) => {
+          const newVersionData =
+            manifestFile.resolutions && versionData.dependencies
```

Comment:
Doesn't this means that we'll add all **root manifest file**'s resolutions to each dependency **as a dependency**?

Please add more complex fixtures and I think you'll see the error